File: common/src/main/java/com/colaman/kyle/common/helper/AnimationHelper.java
package com.colaman.kyle.common.helper;

import android.view.View;

/**
 * Create by kyle on 2019/1/11
 * Function : 动画辅助类
 */
public class AnimationHelper {
    public static void height(float scale, View view, long duration) {
        if (view == null || scale <= 0) {
            return;
        }
        view.animate()
                .scaleYBy(scale)
                .setDuration(duration)
                .start();
    }

    public static void width(float scale, View view, long duration) {
        if (view == null || scale <= 0) {
            return;
        }
        view.animate()
                .scaleXBy(scale)
                .setDuration(duration)
                .start();
    }

    public static void size(float scale, View view, long duration) {
        if (view == null || scale <= 0) {
            return;
        }
        view.animate()
                .scaleXBy(scale)
                .scaleYBy(scale)
                .setDuration(duration)
                .start();
    }

    public static void scrollX(float x, View view, long duration) {
        if (view == null) {
            return;
        }
        view.animate()
                .translationXBy(x)
                .setDuration(duration)
                .start();
    }

    public static void scrollY(float y, View view, long duration) {
        if (view == null) {
            return;
        }
        view.animate()
                .translationYBy(y)
                .setDuration(duration)
                .start();
    }

    public static void scroll(float x, float y, View view, long duration) {
        if (view == null) {
            return;
        }
        view.animate()
                .translationXBy(x)
                .translationXBy(y)
                .setDuration(duration)
                .start();
    }
}

File: base/src/main/java/com/kyle/colman/helper/AnimationHelper.kt
package com.kyle.colman.helper

import android.view.View

/**
 * Create by kyle on 2019/1/11
 * Function : 动画辅助类
 */
object AnimationHelper {
    fun height(scale: Float, view: View?, duration: Long) {
        if (view == null || scale <= 0) {
            return
        }
        view.animate()
            .scaleYBy(scale)
            .setDuration(duration)
            .start()
    }

    fun width(scale: Float, view: View?, duration: Long) {
        if (view == null || scale <= 0) {
            return
        }
        view.animate()
            .scaleXBy(scale)
            .setDuration(duration)
            .start()
    }

    fun size(scale: Float, view: View?, duration: Long) {
        if (view == null || scale <= 0) {
            return
        }
        view.animate()
            .scaleXBy(scale)
            .scaleYBy(scale)
            .setDuration(duration)
            .start()
    }

    fun scrollX(x: Float, view: View?, duration: Long) {
        if (view == null) {
            return
        }
        view.animate()
            .translationXBy(x)
            .setDuration(duration)
            .start()
    }

    fun scrollY(y: Float, view: View?, duration: Long) {
        if (view == null) {
            return
        }
        view.animate()
            .translationYBy(y)
            .setDuration(duration)
            .start()
    }

    fun scroll(x: Float, y: Float, view: View?, duration: Long) {
        if (view == null) {
            return
        }
        view.animate()
            .translationXBy(x)
            .translationXBy(y)
            .setDuration(duration)
            .start()
    }
}